src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java

```diff
@@ -318,79 +305,116 @@ public BaseResponseBean advanceSessionWithSelections(MenuSession menuSession,
             if (notificationMessage == null) {
                 notificationMessage = new NotificationMessage(null, false, NotificationMessage.Tag.selection);
             }
-            BaseResponseBean responseBean = new BaseResponseBean(null, notificationMessage,true);
-            return responseBean;
+            return new BaseResponseBean(null, notificationMessage,true);
         }
     }
 
+    /**
+     * Apply an actions to the menu session that do not require user input e.g.
+     *  - auto launch
+     *  - queries
+     *  - auto advance menu
+     * @param menuSession
+     * @param currentInput The current input being processed
+     * @param nextInput The next input being processed or NO_SELECTION constant
+     * @param queryData Query data from the request
+     * @param needsDetail Whether the full entity screen is required
+     * @param inputValidated Whether the input has been validated (allows skipping validation)
+     * @param forceManualAction Prevent auto execution of queries if true.
+     * @return
+     * @throws CommCareSessionException
+     */
+    private Screen autoAdvanceSession(
```

Comment:
The sync screen is a special case because it's really an `entry` type so we have to treat it as terminal step that rebuilds the session. If it were possible to put sync steps in between other session steps then it would make sense to put that in this method.